app/components/autocomplete/emoji_suggestion/emoji_suggestion.js

```diff
@@ -63,18 +72,19 @@ export default class EmojiSuggestion extends Component {
             return;
         }
 
-        const matchTerm = match[3];
+        const oldMatchTerm = this.matchTerm;
+        this.matchTerm = match[3];
 
-        const matchTermChanged = matchTerm !== this.state.matchTerm;
-        if (matchTermChanged) {
-            this.setState({
-                matchTerm
-            });
+        const pre47 = !isMinimumServerVersion(Client4.getServerVersion(), 4, 7);
```

Comment:
Isn't this stored in the redux store somewhere? It seems weird to access the client directly here.

It also might be nice to have this as a helper function with a clear name to make it clear it's about emojis

Reply:
It is in the redux store, I'll pass it in as a prop